zerver/tests/test_events.py

```diff
@@ -735,26 +736,35 @@ def get_checker(check_gravatar: Validator) -> Validator:
 
     def test_update_message_flags(self) -> None:
         # Test message flag update events
+        message = self.send_personal_message(
+            self.example_email("cordelia"),
+            self.example_email("hamlet"),
+            "hello",
+        )
+        user_profile = self.example_user('hamlet')
+        for first_non_api_flag in UserMessage.NON_API_FLAGS:
+            break
+
+        with self.assertRaises(JsonableError):
+            do_update_message_flags(user_profile, get_client("website"), 'remove', first_non_api_flag, [message])
+
+        with self.assertRaises(JsonableError):
+            do_update_message_flags(user_profile, get_client("website"), 'remove', 'invalid', [message])
+
```

Comment:
`test_events.py` is really just for event system races; we should move your test changes.  `git grep messages/flags` shows that `test_messages.py` is probably where this belongs.

Reply:
@timabbott Can you have a look at `test_messages.py`, I'm not sure under which test case class should this fall? 
(When adding the test initially I had a look at `test_messages.py`, but I didn't find any relevant test case classes.)

Reply:
I'd add a new test function in `MessageAccessTests`, next to the starring tests.